src/edu/csus/ecs/pc2/ui/RunsTablePane.java

```diff
@@ -2146,7 +2224,12 @@ private void showSourceForSelectedRun() {
                         SerializedFile[] otherFiles = runFiles.getOtherFiles();
 
                         // create a MultiFileViewer in which to display the runFiles
-                        MultipleFileViewer mfv = new MultipleFileViewer(log, "Source files for Site " + fetchedRun.getSiteNumber() + " Run " + fetchedRun.getNumber());
+                        // Note: previously used 'fetchedRun' here for site/number; it is possible that those values are not
+                        // correct if the run files are already present; in that case, we would have never asked for them to be
+                        // retrieved, and would have used whatever was in fetchedRun.  An NPE was also possible if fetchedRun was null;
+                        // that is, the runfiles were already present from an edit run or judge run on the current client, not no
+                        // server request was every made for a run's files.
```

Comment:
I can't parse the final clause of the above sentence.  Was it intended to say "`so` no server request was `ever` made"?

Reply:
Yes.  Precisely what you said.  Basically, View Source never made a request to the server for runFiles, then fetchedRun/fetchedRunFiles would be null.  I will change that when the review process is complete.